opus/application/static_media/js/browse.js

```diff
@@ -424,6 +464,50 @@ var o_browse = {
         });
     }, // end browse behaviors
 
+    // update order arrows right away when user clicks on sorting arrows in pill or table header
+    // sync up arrows in both sorting pill and table header
+    updateOrderIndicator: function(headerOrderIndicator, pillOrderIndicator, isDescending, slug) {
+        let headerOrder = isDescending ? "sort-desc" : "sort-asc";
+        let headerOrderArrow = isDescending ? tableSortUpArrow : tableSortDownArrow;
+        let isPillOrderDesc = isDescending ? "true" : "false";
+        let pillOrderTooltip = isDescending ? "Change to ascending sort" : "Change to descending sort";
+        let pillOrderArrow = isDescending ? pillSortUpArrow : pillSortDownArrow;
+
+        // If header already exists, we update the header arrow, else we do nothing
+        if (headerOrderIndicator && headerOrderIndicator.length !== 0) {
+            headerOrderIndicator.data("sort", `${headerOrder}`);
+            headerOrderIndicator.attr("class", `column_ordering ${headerOrderArrow}`);
+
+            // Reset arrows on rest of table headers
+            // let headers = $(`.dataTable th a:not([data-slug="opusid"], [data-slug=${slug}])`).find("span:last");
+            let headers = $(`.dataTable th a:not([data-slug=${slug}])`).find("span:last");
+            headers.data("sort", "none");
+            headers.attr("class", defaultTableSortArrow);
+        }
+
+        // Re-render each pill
+        let listHtml = "";
+        $.each(opus.prefs.order, function(index, orderEntry) {
+            isPillOrderDesc = orderEntry[0] === "-" ? "true" : "false";
+            pillOrderArrow = orderEntry[0] === "-" ? pillSortUpArrow : pillSortDownArrow;
+            orderEntry = orderEntry[0] === "-" ? orderEntry.slice(1) : orderEntry;
+
+            // retrieve label from either displayed header's data-label attribute or displayed pill's text
+            let label = $(`#browse .dataTable th a[data-slug="${orderEntry}"]`).data("label") || $(`#browse .sort-contents span[data-slug="${orderEntry}"] .flip-sort`).text();
+
+            listHtml += "<li class='list-inline-item'>";
+            listHtml += `<span class='badge badge-pill badge-light' data-slug="${orderEntry}" data-descending="${isPillOrderDesc}">`;
+            if (orderEntry !== "opusid") {
```

Comment:
So this is really interesting. We are using different logic here and in updateSortOrder() to decide which pills should have the (x). In this case, you're saying any "opusid" needs to stay. In updateSortOrder(), it is using the "removeable" field from the returned data from dataimages.json. The logic for that field is in results/views.py line 189, where it says that the *last* field shouldn't be removeable, regardless of what it is. If the opusid isn't the last field, these two methods will cause different behaviors, resulting in the weird case of changing the sort order rendering the pill with opusid without an (x), and then when dataimages.json returns re-rendering them with the last field without an (x).

Note: This is a classic example of why it's bad to have mostly duplicated code that does things in different ways!

In this particular case, I think you have exposed a bug in the API. The way we're doing things right now, we must always have opusid *somewhere* in the order list, and thus we should make that one not removeable. Please change results/views.py line 189 from:
`removeable = idx != len(order_slugs)-1;` (I have no idea why this line ends with a ; - that's not a Python thing)
to:
`removeable = not slug.endswith('opusid')`
Don't forget to re-run the tests since you've touched the Django.